### ISSUE TITLE
CLOCK_MONOTONIC  vs gethrtime() HPUX 11.31 issue

### ISSUE THREAD

**Issue by fabe33** [eclipse/mosquitto#805]
HI everybody,

I'm trying to compile libmosquitto.so.1 on an HPUX systeme (HPUX 11iV3/IA64) but i have a problem with this code

```
#elif _POSIX_TIMERS>0 && defined(_POSIX_MONOTONIC_CLOCK)
	struct timespec tp;

	clock_gettime(CLOCK_MONOTONIC, &tp);
	return tp.tv_sec;
```
When i simply replace CLOCK_MONOTONIC by CLOCK_REALTIME , i'm able to compile shared lib libmosquitto.so.1 .
but when i replace this code by gethrtime() the ld linker refuse to create shared libmosquitto.so.1

 Error message ::: shared library must be position independent. Use +z or +Z to recompile.

Can i stay with option CLOCK_REALTIME

**Comment by karlp** [comment]
what do you want to do? you're talking about two different things and I find it hard to follow what you're problem is.

do you have posix_timers and posix_monotoni clock defined, but clock_gettime fails to compile? that seems unlikely, but hey, hpux.

do you just want an extra #else clause for HPUX to use gethrtime instead of the fallback to time() that currently is in place?
